pkg/controller/machinehealthcheck/machinehealthcheck_controller.go

```diff
@@ -368,14 +389,36 @@ func (t *target) remediate(r *ReconcileMachineHealthCheck) error {
 		}
 	}
 	if t.isMaster() {
+		r.recorder.Eventf(
+			&t.Machine,
+			corev1.EventTypeNormal,
+			healthcheckingv1alpha1.EventSkippedMaster,
+			"Machine %v is a master node, skipping remediation",
+			t.Machine.Name,
+		)
 		glog.Infof("%s: master node, skipping remediation", t.string())
 		return nil
 	}
 
 	glog.Infof("%s: deleting", t.string())
 	if err := r.client.Delete(context.TODO(), &t.Machine); err != nil {
+		r.recorder.Eventf(
+			&t.Machine,
+			corev1.EventTypeWarning,
+			healthcheckingv1alpha1.EventMachineDeletionFailed,
+			"Machine %v remediation failed: unable to delete Machine object: %v",
+			t.Machine.Name,
+			err,
+		)
 		return fmt.Errorf("%s: failed to delete machine: %v", t.string(), err)
 	}
+	r.recorder.Eventf(
+		&t.Machine,
```

Comment:
This message might be confusing as is not 100% accurate. Machine is being requested for deletion and the rest of the process is out of band:
- machineSet reconciles towards desired number of replicas and start process to bring a new machine/node.
- machine controller drains node
- machine controller provider deletes instance
- machine controller remove machine resource from disk

May be "Machine %v has been remetiated by requesting to delete Machine object" or similar?

Reply:
I agree. Fixed it.